### PR TITLE
Chore: Encapsulate details sidebar related code

### PR DESCRIPTION
**Pull request by priyajeet**
Move details sidebar code from content sidebar to within details sidebar

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #583

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/ContentSidebar.js [hunk 1]
```diff
@@ -598,43 +391,28 @@ class ContentSidebar extends PureComponent<Props, State> {
...
         return (
-            <Internationalize language={language} messages={intlMessages}>
+            <Internationalize language={language} messages={messages}>
                 <aside id={this.id} className={styleClassName}>
                     <div className="be-app-element">
                         {hasSidebar ? (
                             <APIContext.Provider value={(this.api: any)}>
                                 <Sidebar
                                     file={((file: any): BoxItem)}
+                                    fetchFile={this.fetchFile}
```
Comment (priyajeet) [inline src/components/ContentSidebar/ContentSidebar.js]: remove

File: src/components/ContentSidebar/DetailsSidebar.js [hunk 2]
```diff
@@ -29,89 +37,306 @@ type ExternalProps = {
...
+    };
+
+    /**
+     * Handles a failed file access stats fetch
+     *
+     * @private
+     * @param {Error} e - API error
+     * @return {void}
+     */
+    fetchAccessStatsErrorCallback = (e: $AxiosXHR<any>) => {
```
Comment (jstoffan) [inline src/components/ContentSidebar/DetailsSidebar.js]: Can we rename this to `error` while we're cleaning things up?

File: src/components/ContentSidebar/ContentSidebar.js [hunk 3]
```diff
@@ -219,136 +208,13 @@ class ContentSidebar extends PureComponent<Props, State> {
...
-        const { hasAccessStats = false } = detailsSidebarProps;
-        if (!fileId) {
-            return;
-        }
-
-        // Fetch the new file
-        this.fetchFile(fileId);
-
-        if (hasAccessStats) {
-            this.fetchFileAccessStats(fileId);
```
Comment (jeremypress) [inline src/components/ContentSidebar/ContentSidebar.js]: Will we need to handle a changed file ID in the details sidebar component now? We only fetch access stats on mount.

File: src/components/ContentSidebar/DetailsSidebar.js [hunk 4]
```diff
@@ -29,89 +37,306 @@ type ExternalProps = {
...
+    };
+
+    /**
+     * Add classification click handler
+     *
+     * @private
+     * @return {void}
+     */
+    onClassificationClick = (): void => {
+        const { onClassificationClick = noop }: Props = this.props;
```
Comment (jstoffan) [inline src/components/ContentSidebar/DetailsSidebar.js]: Should this default prop value live at the class level?

File: src/components/ContentSidebar/ContentSidebar.js [hunk 5]
```diff
@@ -219,136 +208,13 @@ class ContentSidebar extends PureComponent<Props, State> {
      * @param {boolean} hasFileIdChanged true if the file id has changed
```
Comment (jeremypress) [inline src/components/ContentSidebar/ContentSidebar.js]: Can you remove this? it's not used.

File: src/components/ContentSidebar/DetailsSidebar.js [hunk 6]
```diff
@@ -29,89 +37,306 @@ type ExternalProps = {
...
+            </SidebarContent>
+        );
+    }
+}
 
 export type DetailsSidebarProps = ExternalProps;
-export default DetailsSidebar;
+export { DetailsSidebar as DetailsSidebarComponent };
+export default (props: PropsWithoutContext) => (
+    <APIContext.Consumer>
```
Comment (jstoffan) [inline src/components/ContentSidebar/DetailsSidebar.js]: Seems like a candidate for a HOC.

File: yarn.lock [hunk 7]
```diff
@@ -5,12 +5,14 @@
 "@babel/code-frame@^7.0.0":
   version "7.0.0"
   resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.0.0.tgz#06e2ab19bdb535385559aabb5ba59729482800f8"
+  integrity sha512-OfC2uemaknXr87bdLUkWog7nYuliM9Ij5HUcajsVcMCpQrcLmtxRbVFTIqmcSkSeYRBFBRxs2FiUqFJDLdiebA==
```
Comment (jstoffan) [inline yarn.lock]: I assume this property is backward-compatible with older versions of yarn?

File: src/components/ContentSidebar/DetailsSidebar.js (hunk 4 above)
Comment (priyajeet) [reply src/components/ContentSidebar/DetailsSidebar.js]: yup

File: yarn.lock (hunk 7 above)
Comment (priyajeet) [reply yarn.lock]: yup

File: src/components/ContentSidebar/DetailsSidebar.js (hunk 6 above)
Comment (priyajeet) [reply src/components/ContentSidebar/DetailsSidebar.js]: thats the goal, withAPI()

File: src/components/ContentSidebar/ContentSidebar.js (hunk 3 above)
Comment (priyajeet) [reply src/components/ContentSidebar/ContentSidebar.js]: That should auto happen due to key="fileid" in Sidebar.js
Eventually though, that will have to be replaced by file version id

**Pull request merged**